PrePi/FdtParser.c

```diff
@@ -91,6 +118,67 @@ FindMemnodeFdt (
     }
   }
 
+  //
+  // no-map and no-map-fixup
+  //
+  HighestNoMap = 0;
+  AddressCells = (AddressCellsRoot == -1) ? 1 : AddressCellsRoot;

```

Comment:
if the root node wasn't found you'll override AddressCells to 1 even though it's either still 1 or got set by the local node.
If the root node was found you use AddressCellsRoot no matter if the local one has been found or not.

Reply:
Line 134-137 overrides it if a local value is found in the other node checked